### ISSUE TITLE
Compilation Issue on Raspberry Pi

### ISSUE THREAD

**Issue by o3o** [andrewlalis/handy-httpd#34]
First of all I wanted to express my appreciation for your excellent library. 

I've been trying to use it for a project involving a Raspberry Pi, but unfortunately, I encountered a compilation issue.

**Error Message**
```
> dub build
    Starting Performing "debug" build using /usr/bin/ldc2 for arm, arm_hardfloat.
    Building handy-httpd 8.3.0: building configuration [library]
'-sse4.2' is not a recognized feature for this target (ignoring feature)
SSE: false
source/handy_httpd/components/multivalue_map.d(251,27): Error: cannot implicitly convert expression `idx` of type `long` to `uint`
source/handy_httpd/components/multivalue_map.d(137,46): Error: cannot implicitly convert expression `idx` of type `long` to `uint`
source/handy_httpd/components/multipart.d(251,19): Error: cannot implicitly convert expression `index` of type `long` to `uint`
source/handy_httpd/components/multipart.d(251,28): Error: cannot implicitly convert expression `index + cast(long)subset.length` of type `long` to `uint`
source/handy_httpd/components/multipart.d(98,31): Error: template instance `handy_httpd.components.multipart.indexOf!(immutable(char))` error instantiating
source/handy_httpd/components/multipart.d(102,35): Error: cannot implicitly convert expression `boundaryIdx + 9L` of type `long` to `uint`
source/handy_httpd/components/multipart.d(134,25): Error: cannot implicitly convert expression `nextIdx` of type `long` to `uint`
source/handy_httpd/components/multipart.d(138,39): Error: cannot implicitly convert expression `nextIdx` of type `long` to `uint`
source/handy_httpd/components/multipart.d(138,50): Error: cannot implicitly convert expression `nextIdx + cast(long)boundary.length + 4L` of type `long` to `uint`
source/handy_httpd/components/multipart.d(146,49): Error: cannot implicitly convert expression `elementStartIdx` of type `const(ulong)` to `uint`
source/handy_httpd/components/multipart.d(146,68): Error: cannot implicitly convert expression `elementEndIdx` of type `const(ulong)` to `uint`
source/handy_httpd/components/multipart.d(167,31): Error: cannot implicitly convert expression `bodyIdx` of type `ulong` to `uint`
source/handy_httpd/components/multipart.d(189,37): Error: cannot implicitly convert expression `nextHeaderIdx` of type `ulong` to `uint`
source/handy_httpd/components/multipart.d(193,37): Error: cannot implicitly convert expression `nextHeaderIdx` of type `ulong` to `uint`
source/handy_httpd/components/multipart.d(193,54): Error: cannot implicitly convert expression `headerEndIdx` of type `const(long)` to `uint`
source/handy_httpd/components/multipart.d(196,64): Error: cannot implicitly convert expression `nextHeaderIdx` of type `ulong` to `uint`
source/handy_httpd/components/multipart.d(196,81): Error: cannot implicitly convert expression `nextHeaderIdx + 2LU` of type `ulong` to `uint`
source/handy_httpd/components/multipart.d(206,54): Error: cannot implicitly convert expression `headerValueSeparatorIdx` of type `const(long)` to `uint`
Error /usr/bin/ldc2 failed with exit code 1.

```

**Environment**
Raspberry Pi Model: [Specify the model]
OS: Linux 5.10.60-1-ARCH #1 SMP Sat Aug 21 14:56:38 UTC 2021 armv7l GNU/Linux
Compiler: LDC - the LLVM D compiler (1.36.0):
Library Version: 8.3.0

**Steps to Reproduce**
```
> git clone https://github.com/andrewlalis/handy-httpd.git
> cd handy-httpdj
> dub build
```

thank you

**Comment by o3o** [comment]
Thank you very much.
Now compiler complains:
```
source/handy_httpd/components/websocket/frame.d(244,32): Error: cannot implicitly convert expression `payloadLength` of type `ulong` to `uint`
source/handy_httpd/components/websocket/frame.d(175,33): Error: template instance `handy_httpd.components.websocket.frame.readPayload!(SocketInputStream*)` error instantiating
source/handy_httpd/components/websocket/manager.d(191,42):        instantiated from here: `receiveWebSocketFrame!(SocketInputStream)`
```

see my PR.

Thank you again

**Comment by andrewlalis** [comment]
Additional error messages should be fixed in v8.3.2. Please reopen if there are more issues with ARM, as, like I said, I can't really test it myself, nor does github actions offer arm runners.
